tests/controller_test.py

```diff
@@ -8,65 +9,79 @@
 
 print(evalit.__version__)
 
+from datetime import datetime
+from typing import List, Tuple
+
 # read a yaml file
 import matplotlib.pyplot as plt
 import numpy as np
-import yaml
+from loguru import logger
+
+from evalit.api import MFTAutomation, NifiAutomation, RcloneAutomation, TransferDTO
 
-# from mft.mft_automation import MFTAutomation
-# from nifi.nifi_automation import NifiAutomation
-print(dir(evalit))
-from evalit.rclone import RcloneAutomation
 
+def dtotimes_to_times(timesdto: Tuple[TransferDTO]) -> List[List[float]]:
+    times = map(
+        lambda d: [
+            (d.start_time - datetime(1970, 1, 1)).total_seconds(),
+            (d.end_time - datetime(1970, 1, 1)).total_seconds(),
+        ],
+        timesdto,
+    )
+    return list(times)
 
-def generate_grapgs(title, times):
-    times = times - np.min(times)
+
+def generate_grapgs(title: str, timesdto: Tuple[TransferDTO]):
+    times = filter(
+        lambda t: t.start_time is not None and t.end_time is not None, timesdto
+    )
+    times = tuple(times)
+    times = dtotimes_to_times(times)
 
     for i in range(len(times)):
         plt.barh(i + 1, times[i][1] - times[i][0], left=times[i][0])
 
     plt.savefig(title + ".png")
 
 
-def caclulate_throughput(file_sizes, times):
+def caclulate_throughput(file_sizes: List[int], timesdto: Tuple[TransferDTO]) -> float:
+    timesdto = filter(
+        lambda t: t.start_time is not None and t.end_time is not None, timesdto
+    )
+    times = dtotimes_to_times(timesdto)
     total_valume = 0.0
     for i in range(len(file_sizes)):
         total_valume += file_sizes[i]
-    throughput = total_valume * 8 / (np.max(times) - np.min(times))
-    print("Throughput: " + str(throughput))
+    val = 0
+    try:
+        val = total_valume * 8 / (np.max(times) - np.min(times))
+    except ZeroDivisionError:
+        logger.error("Error while computing throughput!")
+        val = 0
+    return val
 
 
 # TODO: Fetch theses values from the S3 python client
 file_sizes = [1]  # GB
 file_list = ["testfile"]  # File list in source bucket
 
 # nifi_installation = "/sproj/MFT/nifi-1.15.3"
-nifi_installation = "/home/nishan/software/nifi/nifi-1.15.3"
-mft_installation = "/proj/MFT/build"
+nifi_installation = os.getenv(
+    "NIFI_INSTALLATION", "/home/nishan/software/nifi/nifi-1.15.3"
```

Comment:
Can we avoid this hardcoding?

Reply:
Thanks. Will do. I wanted to add fallback value for my device as a test.